src/components/progress-summary/progress-extractor.service.ts

```diff
@@ -65,15 +58,103 @@ export class ProgressExtractor {
     return null;
   }
 
-  extractReportPeriod(pnp: WorkBook, file: FileVersion): Progress | null {
-    // TODO implement extraction
+  extractReportPeriod(
+    pnp: WorkBook,
+    file: FileVersion,
+    fiscalQuarter: number,
+    fiscalYear: number
+  ): Progress | null {
+    const sheet = pnp.Sheets.Progress;
+    if (!sheet) {
+      this.logger.warning('Unable to parse pnp file', {
+        name: file.name,
+        id: file.id,
+      });
+      return null;
+    }
+
+    const rows = utils.sheet_to_json<any>(sheet, {
+      header: 'A',
+      raw: false,
+    });
+
+    try {
+      for (const row of rows) {
+        if (fiscalYear === parseInt(row.AG)) {
+          if (parseInt(row.AH) === fiscalQuarter) {
+            return this.parseRawData(row.AH, row.AH);
+          } else if (parseInt(row.AI) === fiscalQuarter) {
+            return this.parseRawData(row.AI, row.AI);
+          } else if (parseInt(row.AJ) === fiscalQuarter) {
+            return this.parseRawData(row.AJ, row.AJ);
+          } else if (parseInt(row.AK) === fiscalQuarter) {
+            return this.parseRawData(row.AK, row.AK);
+          }
+        }
+      }
+    } catch (e) {
```

Comment:
@blakekrammes what is this catch for?

Reply:
`parsePercent`, which is called by `parseRawData`, can throw an error.